### DISCUSSION TITLE
RoomState 외에 데이터를 저장하는법

### DISCUSSION THREAD

**Discussion by JeongYire** [naverz/zepeto-studio-kor#331] [5. Q&A]
Room 안에서만 활용되는 서버단의 데이터를 다루고싶습니다
하지만 State 로 제어를하면 OnStateChange 함수가 계속 발동이 되는지라. 
해봐야 두세번밖에 발동하지않을 데이터지만 유저끼리 데이터가 동기화가 되어야하기때문에 State 같은 데이터  저장방법이 필요합니다.
OnStateChange 를 발동시키지 않고도 Room내에서만 활용될 데이터를 저장할 수 있는 방법이 있을까요? 
서버파일내의 index.ts 안에 그냥 아예 변수를 선언하는방법은 시도해보진 않았지만 아마 안될것같구요.

**Comment by moondory77-naverz** [comment]
안녕하세요.
Room State는 서버-클라이언트 간의 데이터 동기화를 위한 **최상단** Schema입니다.
또한, OnStateChange는 이에 상응하는 **최상단** 이벤트 리스너입니다.

때문에 동기화가 필요한 데이터임에도, 데이터가 갱신되었을 때
OnStateChange 콜백이 호출되지 않도록 하는 것은 불가능합니다,

감사합니다.
